### ISSUE TITLE
Emitter.stream: Is there a way to catch exceptions ?

### ISSUE THREAD

**Issue by gabime** [liangxianzhe/creator#25]
Consider the following stream with exception:

```dart

Stream<int> someStream() async* {
  while (true) {
    await Future.delayed(const Duration(seconds: 1));
    throw Exception("Test exception");
    yield 123;
  }
}

final someStreamEmitter = Emitter.stream((ref) => someStream());

```

This will produce "[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception"

Is there a way to handle exceptions in stream ? Seems the [current implementation ](https://github.com/liangxianzhe/creator/blob/2c03b97b3bca0c97456f28ce1121d0fa023695d8/packages/creator_core/lib/src/creator.dart#L193) is not passing the `onError` param to `listen()`

**Comment by liangxianzhe** [comment]
You are right. We need a better approach on this. I just merged a fix https://github.com/liangxianzhe/creator/commit/e7de5b2be151825c7e7de205e16d2595766c6b73 and published a new version `0.3.2`. Tested on the below code, but let me know if you still see issue on have a better approach on the fix.

```dart
import 'package:creator_core/creator_core.dart';

Stream<int> someStream() async* {
  while (true) {
    await Future.delayed(const Duration(seconds: 1));
    throw Exception("Test exception");
    yield 123;
  }
}

Future<void> main(List<String> args) async {
  final ref = Ref();
  final someStreamEmitter = Emitter.stream((ref) => someStream());
  try {
    print(await ref.watch(someStreamEmitter));
  } catch (e) {
    print("caught $e");
  }
  print("done");
}
```

**Comment by gabime** [comment]
Great. Thanks for the quick fix!
